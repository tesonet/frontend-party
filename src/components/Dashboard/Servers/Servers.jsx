import React, { Component } from "react";
import axios from "axios";
import _ from "lodash";

import Config from "utils/Config";
import ServerListItem from "./ServersListItem";
import styles from "./Servers.module";

export default class Servers extends Component {
  constructor(props) {
    super(props);
    this.state = {
      servers: [],
      isLoading: true,
      errors: null,
      sortAsc: true
    };
    this.sortBy.bind(this);
  }

  getServers() {
    var token = localStorage.getItem("testioToken");

    axios
      .get(Config.REST_API_URL + "servers", {
        headers: {
          Authorization: token,
          "Content-Type": "application/json"
        }
      })
      .then(response => {
        let servers = response.data.map(item => ({
          id: item.name,
          name: item.name,
          distance: item.distance
        }));
        let serversOrdered = _.orderBy(
          servers,
          ["name", "distance"],
          ["asc", "asc"]
        );
        this.setState({
          servers: serversOrdered,
          isLoading: false
        });
      })
      .catch(error => this.setState({ error, isLoading: false }));
  }

  sortBy(key) {
    var serversCopy = _.orderBy(
      this.state.servers,
      [key],
      this.state.sortAsc ? "asc" : "desc"
    );
    this.setState({ sortAsc: !this.state.sortAsc });
    this.setState({ servers: serversCopy });
  }

  componentDidMount() {
    this.getServers();
  }
  render() {
    const { isLoading, servers } = this.state;

    return (
      <table className={styles.dataTable}>
        <thead>
          <tr>
            <th
              className={styles.dataTable__name}
              onClick={() => this.sortBy("name")}
            >
              Server
            </th>
            <th
              className={styles.dataTable__distance}
              onClick={() => this.sortBy("distance")}
            >
              Distance
            </th>
          </tr>
        </thead>
        <tbody>
          {!isLoading ? (
            servers.map((item, index) => {
              return (
                <ServerListItem
                  key={index}
                  name={item.name}
                  distance={item.distance}
                />
              );
            })
          ) : (
            <tr>
              <td colSpan="2" className={styles.dataTable__loading}>
                Loading...
              </td>
            </tr>
          )}
        </tbody>
      </table>
    );
  }
}
