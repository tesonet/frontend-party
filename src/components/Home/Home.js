import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import HomeTableToolbar from '../HomeHeader/HomeHeader'
import HomeTableHead from './HomeTableHead/HomeTableHead'
import styles from './styles'
import actions from './actions'

export default function Home() {
    const classes = styles();
    const {
        rows,
        selected,
        rowsPerPage,
        page,
        order,
        orderBy,
        handleRequestSort,
        handleSelectAllClick,
        stableSort,
        getSorting
    } = actions();

    const isSelected = name => selected.indexOf(name) !== -1;
    const emptyRows = rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);

    return (
        <div className={classes.root}>
            <Paper className={classes.paper}>
                <HomeTableToolbar numSelected={selected.length}/>
                <TableContainer>
                    <Table
                        className={classes.table}
                        aria-labelledby="tableTitle"
                        size={'medium'}
                        aria-label="enhanced table"
                    >
                        <HomeTableHead
                            classes={classes}
                            order={order}
                            orderBy={orderBy}
                            onSelectAllClick={handleSelectAllClick}
                            onRequestSort={handleRequestSort}
                            rowCount={rows.length}
                        />
                        <TableBody>
                            {stableSort(rows, getSorting(order, orderBy))
                                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                                .map((row, index) => {
                                    const isItemSelected = isSelected(row.name);
                                    const labelId = `enhanced-table-checkbox-${index}`;

                                    return (
                                        <TableRow
                                            hover
                                            role="checkbox"
                                            aria-checked={isItemSelected}
                                            tabIndex={-1}
                                            key={row.key}
                                            selected={isItemSelected}
                                        >

                                            <TableCell component="th" id={labelId} scope="row" className={classes.cell}>
                                                {row.name}
                                            </TableCell>
                                            <TableCell align="right">{row.distance}</TableCell>
                                        </TableRow>
                                    );
                                })}
                            {emptyRows > 0 && (
                                <TableRow style={{height: (55) * emptyRows}}>
                                    <TableCell colSpan={6}/>
                                </TableRow>
                            )}
                        </TableBody>
                    </Table>
                </TableContainer>
            </Paper>
        </div>
    );
}