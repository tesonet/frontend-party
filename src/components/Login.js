import React, { Component } from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { auth } from "../store/actions/actions";
import { Redirect } from "react-router-dom";
import image from "../assets/images/riding-the-waves-3440x1440.jpg";
import { FaUser, FaLock } from "react-icons/fa";
import logo from "../assets/images/logo-testio.png";
import Form from "./styles/FormStyles";

const Wrapper = styled.div`
  align-items: center;
  justify-content: center;
  display: flex;
  height: 100vh;
  flex-direction: column;
  background: linear-gradient(rgba(25, 35, 62, 0.8), rgba(25, 35, 62, 0.8)),
    url(${image});
  background-size: cover;
  background-repeat: no-repeat;

  .wrapper-logo {
    margin-bottom: 50px;
  }
`;

class Login extends Component {
  state = {
    username: "",
    password: "",
    toServerList: false
  };

  changeHandler = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  submitHandler = async e => {
    e.preventDefault();

    await this.props.onAuth(this.state.username, this.state.password);
    if (this.props.token) {
      this.setState({ toServerList: true });
    }
  };

  render() {
    let errorMessage = null;
    if (this.props.error) {
      errorMessage = <p className="form-error-message">{this.props.error}</p>;
    }

    if (this.state.toServerList) {
      return <Redirect to="/server-list" />;
    }

    return (
      <Wrapper>
        <img className="wrapper-logo" src={logo} alt="Testio Logo" />
        <Form onSubmit={this.submitHandler}>
          <fieldset disabled={this.props.loading}>
            {errorMessage}
            <div className="input-container">
              <FaUser
                style={{
                  color: "#cccccc",
                  background: "white",
                  position: "absolute",
                  left: "13px",
                  top: "20px",
                  minWidth: "30px",
                  textAlign: "center"
                }}
              />
              <input
                required
                type="text"
                name="username"
                placeholder="Username"
                value={this.username}
                onChange={this.changeHandler}
              />
            </div>

            <div className="input-container">
              <FaLock
                style={{
                  color: "#cccccc",
                  background: "white",
                  position: "absolute",
                  left: "13px",
                  top: "20px",
                  minWidth: "30px",
                  textAlign: "center"
                }}
              />
              <input
                required
                type="password"
                name="password"
                placeholder="Password"
                value={this.password}
                onChange={this.changeHandler}
              />
            </div>

            <button className="form-button">Log In</button>
          </fieldset>
        </Form>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => {
  return {
    isAuthenticated: state.token !== null,
    loading: state.loading,
    error: state.error,
    token: state.token
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onAuth: (username, password) => dispatch(auth(username, password))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
