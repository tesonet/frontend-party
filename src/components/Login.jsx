import React from 'react';
import { withAlert } from 'react-alert';

import Icon from './Icon';
import InputGroup from './InputGroup';
import TesonetAPI from '../utilities/TesonetAPI';
import bindFunctions from '../utilities/bindFunctions';

import bigLogo from '../images/bigLogo.png';

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      password: '',
      username: '',
      loggingIn: false
    };

    bindFunctions(
      this,
      'onUsernameChange',
      'onPasswordChange',
      'onSubmit'
    );
  }

  onUsernameChange(event) {
    this.setState({username: event.target.value});
  }

  onPasswordChange(event) {
    this.setState({password: event.target.value});
  }

  onSubmit(event) {
    const {onLogin, alert} = this.props;

    event.preventDefault();

    this.setState({loggingIn: true});

    TesonetAPI
      .createToken(this.state.username, this.state.password)
      .then(
        onLogin,
        error => {
          this.setState({loggingIn: false});
          if (error instanceof TesonetAPI.HTTPError) {
            alert.error('Could not log in with provided username and password.');
          } else {
            alert.error('A network error has occured.');
          }
        }
      );
  }

  render() {
    const {username, password, loggingIn} = this.state;

    const submitDisabled = username === '' || password === '' || loggingIn;

    return (
      <div className="login-wrapper">
        <div className="container">
          <div className="login-container">
            <div className="login-inner">
              <img src={bigLogo} alt="testio." />
              <form onSubmit={this.onSubmit}>
                <InputGroup>
                  <Icon icon="fa-user" title="Username" />
                  <input
                    value={username}
                    onChange={this.onUsernameChange}
                    type="text"
                    placeholder="Username"
                    aria-label="Username"
                    className="form-control"
                  />
                </InputGroup>
                <InputGroup>
                  <Icon icon="fa-lock" title="Password" />
                  <input
                    value={password}
                    onChange={this.onPasswordChange}
                    type="password"
                    placeholder="Password"
                    aria-label="Password"
                    className="form-control"
                  />
                </InputGroup>
                <button type="submit" disabled={submitDisabled}>
                  {loggingIn ? 'Logging Inâ€¦' : 'Log In'}
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default withAlert(Login);
