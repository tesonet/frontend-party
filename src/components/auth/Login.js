import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import LoginLayout from './LoginLayout';
import FormField from '../utils/form/FormField';
import FormButton from '../utils/form/FormButton';
import {
    updateField,
    generateData,
    isFormValid
} from '../utils/form/formActions';
import { login } from '../../actions/authActions';

class Login extends Component {
    state = {
        formError: false,
        formErrorMessage: '',
        formdata: {
            username: {
                element: 'input',
                value: '',
                config: {
                    name: 'username',
                    type: 'text',
                    placeholder: 'Username'
                },
                icon: true,
                validation: {
                    required: true
                },
                valid: false,
                touched: false,
                errorMessage: ''
            },
            password: {
                element: 'input',
                value: '',
                config: {
                    name: 'password',
                    type: 'password',
                    placeholder: 'Password'
                },
                icon: true,
                validation: {
                    required: true
                },
                valid: false,
                touched: false,
                errorMessage: ''
            }
        }
    };

    updateForm = element => {
        const newFormdata = updateField(element, this.state.formdata);
        this.setState({
            formError: false,
            formErrorMessage: '',
            formdata: newFormdata
        });
    };

    onSubmit = e => {
        e.preventDefault();

        let dataToSubmit = generateData(this.state.formdata);
        let formIsValid = isFormValid(this.state.formdata);

        if (formIsValid) {
            this.props.login(dataToSubmit).then(() => {
                if (this.props.auth.message) {
                    this.setState({
                        formError: true,
                        formErrorMessage: this.props.auth.message
                    });
                }
            });
        } else {
            this.setState({
                formError: true,
                formErrorMessage: 'Something went wrong'
            });
        }
    };

    renderError = () =>
        this.state.formError && (
            <div className="form__error">{this.state.formErrorMessage}</div>
        );

    render() {
        if (this.props.auth.token) return <Redirect to="/servers" />;

        return (
            <LoginLayout>
                <form onSubmit={e => this.onSubmit(e)} className="form">
                    <FormField
                        id={'username'}
                        fieldData={this.state.formdata.username}
                        change={element => this.updateForm(element)}
                    />

                    <FormField
                        id={'password'}
                        fieldData={this.state.formdata.password}
                        change={element => this.updateForm(element)}
                    />

                    <FormButton type="submit" value="Log in" />
                    {this.renderError()}
                </form>
            </LoginLayout>
        );
    }
}

const mapStateToProps = state => ({
    auth: state.auth
});

export default connect(
    mapStateToProps,
    { login }
)(Login);
