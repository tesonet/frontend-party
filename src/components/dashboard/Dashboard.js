import React, { Component } from 'react';
import Navbar from '../layout/Navbar';
import ServerList from '../servers/ServerList';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

class Dashboard extends Component {
  render() {
    const { authToken } = this.props;
    if (authToken === null) return <Redirect to='/signin' />

    return (
      <div className="dashboard">
        <Navbar />
        <ServerList />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    authToken: state.auth.authToken
  }
}

export default connect(mapStateToProps)(Dashboard);