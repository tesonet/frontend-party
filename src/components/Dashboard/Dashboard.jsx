import React, { Component } from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

import Header from "./Header/Header";
import Servers from "./Servers/Servers";

export default class Dashboard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <React.Fragment>
        <ReactCSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnterTimeout={600}
          transitionLeaveTimeout={200}
          transitionName={
            this.props.match.path === "/dashboard" ? "SlideOut" : "SlideIn"
          }
        >
          <Header props={this.props} />
          <Servers />
        </ReactCSSTransitionGroup>
      </React.Fragment>
    );
  }
}
