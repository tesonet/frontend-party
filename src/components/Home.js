import React, { useState, useEffect } from "react";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {
  CssBaseline,
  AppBar,
  makeStyles,
  Toolbar,
  Button
} from "@material-ui/core";
import { logoutAction } from "../actions/auth";
import MaterialTable from "material-table";
import { getServerListAction } from "../actions/servers";

const useStyles = makeStyles(theme => ({
  appBar: {
    borderBottom: `1px solid #e6e6e6`,
    height: 110,
    justifyContent: "center"
  },
  logout: {
    fontWeight: 300,
    textTransform: "none",
    color: "#2f3254",
    "&:hover": {
      backgroundColor: "#99cc33"
    }
  },
  toolbar: {
    flexWrap: "wrap"
  },
  grow: {
    flexGrow: 1
  },
  leftIcon: {
    marginRight: theme.spacing(1)
  }
}));

const options = {
  sorting: true,
  toolbar: false,
  paging: false,
  headerStyle: {
    paddingBottom: 10,
    backgroundColor: "#f5f5f5",
    color: "#999",
    fontWeight: 300,
    letterSpacing: 1.4,
    height: 60
  },
  rowStyle: {
    height: 60
  }
};

const columns = [
  {
    title: "SERVER",
    field: "name",
    cellStyle: {
      color: "#666",
      fontWeight: 300,
      letterSpacing: 0.4,
      borderBottom: `1px solid #e6e6e6`
    }
  },
  {
    title: "DISTANCE",
    field: "distance",
    customSort: (a, b) => a.distanceOriginal - b.distanceOriginal,
    cellStyle: {
      textAlign: "right",
      color: "#666",
      fontWeight: 300,
      letterSpacing: 0.4,
      borderBottom: `1px solid #e6e6e6`
    },
    headerStyle: {
      textAlign: "right"
    },
    defaultSort: "asc"
  }
];

function Home({ logoutAction, getServerListAction, serverList }) {
  useEffect(() => {
    getServerListAction();
  }, [getServerListAction]);

  const [logout, setLogout] = useState(false);
  const classes = useStyles();

  const onClickLogout = () => {
    logoutAction();
    setLogout(true);
  };

  return logout ? (
    <Redirect to="/" />
  ) : (
    <React.Fragment>
      <CssBaseline />
      <AppBar
        position="sticky"
        color="inherit"
        elevation={0}
        className={classes.appBar}
      >
        <Toolbar className={classes.toolbar}>
          <img
            src={require("../resources/testio-black.svg")}
            alt="Testio logo"
            className={classes.logo}
          />
          <div className={classes.grow} />
          <Button onClick={onClickLogout} className={classes.logout}>
            <img
              alt="Logout icon"
              className={classes.leftIcon}
              src={require("../resources/logout.svg")}
            />
            Logout
          </Button>
        </Toolbar>
      </AppBar>
      <MaterialTable
        isLoading={serverList.data.length === 0}
        columns={columns}
        options={options}
        data={serverList.data.map(row => ({
          name: row.name,
          distance: row.distance + " km",
          distanceOriginal: row.distance
        }))}
      />
    </React.Fragment>
  );
}

const mapStateToProps = state => ({
  serverList: state.serverListReducer.serverList
});

const mapDispatchToProps = {
  logoutAction,
  getServerListAction
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);

Home.propTypes = {
  logoutAction: PropTypes.function,
  getServerListAction: PropTypes.function,
  serverList: PropTypes.object
};
