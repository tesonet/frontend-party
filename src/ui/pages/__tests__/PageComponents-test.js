import React from 'react';

import { shallow } from 'enzyme';

// Components
import Home from '../home';
import Dashboard from '../dashboard';
import Login from '../login';
import NotFound from '../notFound';
import Servers from '../dashboard/servers';

describe('Page UI components', () => {

    describe('Home page component', () => {
        const homePageComponent = shallow(<Home/>);
        test('renders correctly', () => {
            expect(homePageComponent).toMatchSnapshot();
        });
    });

    describe('Login page component', () => {
        const loginpageComponent = shallow(<Login />);
        test('renders correctly', () => {
            expect(loginpageComponent).toMatchSnapshot();
        });
    });

    describe('NotFound page container component', () => {
        const notFoundPageComponent = shallow(<NotFound />);
        test('renders correctly', () => {
            expect(notFoundPageComponent).toMatchSnapshot();
        });
    });

    describe('Dashboard pages', () => {
        describe('Dashboard Home page component', () => {
            const dashboardHomePageComponent = shallow(<Dashboard />);
            test('renders correctly', () => {
                expect(dashboardHomePageComponent).toMatchSnapshot();
            });
        });

        describe('Dashboard Servers page component', () => {
            const dashboardServersPageComponent = shallow(<Servers />);
            test('renders correctly', () => {
                expect(dashboardServersPageComponent).toMatchSnapshot();
            });
        });
    });

});
