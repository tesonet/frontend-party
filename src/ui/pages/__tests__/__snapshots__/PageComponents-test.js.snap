// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page UI components Dashboard pages Dashboard Home page component renders correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/dashboard"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/dashboard/servers"
  />
  <Route
    component={[Function]}
    path="/dashboard"
  />
</Switch>
`;

exports[`Page UI components Dashboard pages Dashboard Servers page component renders correctly 1`] = `
<div>
  <h1>
    Servers
  </h1>
</div>
`;

exports[`Page UI components Home page component renders correctly 1`] = `
<Redirect
  push={true}
  to="/dashboard"
/>
`;

exports[`Page UI components Login page component renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Page UI components NotFound page container component renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
